gradle-plugin/src/gradleTest/java/com/squareup/anvil/plugin/AnvilVersionMatrix.kt

```diff
@@ -0,0 +1,33 @@
+package com.squareup.anvil.plugin
+
+import com.rickbusarow.kase.gradle.AgpDependencyVersion
+import com.rickbusarow.kase.gradle.DaggerDependencyVersion
+import com.rickbusarow.kase.gradle.GradleDependencyVersion
+import com.rickbusarow.kase.gradle.HasKotlinDependencyVersion
+import com.rickbusarow.kase.gradle.KotlinDependencyVersion
+import com.rickbusarow.kase.gradle.KspDependencyVersion
+import com.rickbusarow.kase.gradle.VersionMatrix
+
+class AnvilVersionMatrix : VersionMatrix by VersionMatrix(agp + kotlin + gradle + dagger) {
+  companion object {
+    val agp = listOf("7.3.1", "7.4.2", "8.0.2", "8.1.1", "8.2.0").map(::AgpDependencyVersion)
+    val kotlin = listOf("1.8.21", "1.9.0", "1.9.10", "1.9.21").map(::KotlinDependencyVersion)
+    val gradle = listOf("8.5").map(::GradleDependencyVersion)
+    val dagger = listOf("2.46.1").map(::DaggerDependencyVersion)
```

Comment:
You make good points.  I have plans for this but didn't document them.  I'll address that.

For posterity:

These should be independent of the current matrices in `ci.yml`.  Those are more limited since they change what the project builds with, whereas this matrix only defines what the consuming project uses.

Soon, I will need to move this matrix definition into `build-logic/conventions`, so that it can be broken into shards for CI.  The idea is that `ci.yml` is auto-updated via a task, and this specific `AnvilVersionMatrix.kt` in `:gradle-plugin` would be generated like a BuildConfig file.  This creates a single source of truth (in `build-logic/conventions`), keeps CI scalable, and makes it simple to run the full matrix on dev machines.